api/services/monitoring/src/actions/StatsRefreshAction.ts

```diff
@@ -0,0 +1,80 @@
+import { handler } from '@ilos/common';
+import { PoolClient, PostgresConnection } from '@ilos/connection-postgres';
+import { Action as AbstractAction } from '@ilos/core';
+import { internalOnlyMiddlewares } from '@pdc/provider-middleware/dist';
+import { handlerConfig, ParamsInterface, ResultInterface } from '../shared/monitoring/statsrefresh.contract';
+import { alias } from '../shared/monitoring/statsrefresh.schema';
+
+const CronFrequencies = ['daily', 'weekly', 'monthly'] as const;
+type CronFrequency = typeof CronFrequencies[number];
+
+@handler({
+  ...handlerConfig,
+  middlewares: [...internalOnlyMiddlewares('proxy'), ['validate', alias]],
+})
+export class StatsRefreshAction extends AbstractAction {
+  private cn: PoolClient;
+
+  constructor(protected pg: PostgresConnection) {
+    super();
+  }
+
+  public async handle({ schema }: ParamsInterface): Promise<ResultInterface> {
+    try {
+      this.cn = await this.pg.getClient().connect();
```

Comment:
Comme la connexion n'est pas gardée j'aurais tendance à ne pas la mettre dans une ppté de la classe. Genre
```suggestion
      const cn = await this.pg.getClient().connect();
```